### PR TITLE
Do not use sorted sets for authors or declared licenses

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

**Comment by sschuberth** [comment]
This continues the story started in https://github.com/oss-review-toolkit/ort/pull/6244.

File: analyzer/src/main/kotlin/managers/Pub.kt [hunk 1]
```diff
@@ -473,7 +473,7 @@ class Pub(
                     var rawName = ""
                     var homepageUrl = ""
                     var vcs = VcsInfo.EMPTY
-                    var authors: SortedSet<String> = sortedSetOf<String>()
+                    var authors = mutableSetOf<String>()
```
Comment (mnonnenmacher) [inline analyzer/src/main/kotlin/managers/Pub.kt:476]: Should be `emptySet()`.

File: analyzer/src/main/kotlin/managers/Pub.kt [hunk 2]
```diff
@@ -672,8 +672,8 @@ class Pub(
 /**
  * Extract information about package authors from the given [pubspec].
  */
-private fun parseAuthors(pubspec: JsonNode): SortedSet<String> =
-    (listOfNotNull(pubspec["author"]) + pubspec["authors"]?.toList().orEmpty()).mapNotNullTo(sortedSetOf()) {
+private fun parseAuthors(pubspec: JsonNode): MutableSet<String> =
```
Comment (mnonnenmacher) [inline analyzer/src/main/kotlin/managers/Pub.kt:675]: Should be `Set<String>`.

File: analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt [hunk 3]
```diff
@@ -199,20 +199,20 @@ private fun String.sanitize(): String = replace(':', ' ').collapseWhitespace()
...
 
         // Do not split an organization like "Acme, Inc." by comma.
         withoutPrefix(SpdxConstants.ORGANIZATION)?.let {
-            return sortedSetOf(it)
+            return setOf(it)
         }
     }
 
-    return sortedSetOf()
+    return setOf()
```
Comment (mnonnenmacher) [inline analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt:215]: `emptySet()`

File: utils/ort/src/main/kotlin/DeclaredLicenseProcessor.kt [hunk 4]
```diff
@@ -172,5 +172,10 @@ data class ProcessedDeclaredLicense(
      * The list of all mapped and unmapped licenses.
      */
     @JsonIgnore
-    val allLicenses = spdxExpression?.decompose().orEmpty().map { it.toString() } + unmapped
+    val allLicenses = decompose().map { it.toString() } + unmapped
+
+    /**
+     * Decompose the SPDX expression or return an empty string.
```
Comment (mnonnenmacher) [inline utils/ort/src/main/kotlin/DeclaredLicenseProcessor.kt:184]: `[Decompose][SpdxExpression.decompose] the [spdxExpression] or return an empty string if it is null.`

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

**Review by mnonnenmacher — APPROVED** [review]

**Pull request merged**